common/src/main/java/com/viaversion/viarewind/api/type/chunk/BulkChunkType1_7_6.java

```diff
@@ -28,75 +12,65 @@
-		final int[] chunkZ = new int[chunkCount];
-		final short[] primaryBitMask = new short[chunkCount];
-		final short[] additionalBitMask = new short[chunkCount];
+    @Override
+    public void write(ByteBuf byteBuf, Chunk[] chunks) {
+        final int chunkCount = chunks.length;
+        final ByteArrayOutputStream output = new ByteArrayOutputStream();
+        final int[] chunkX = new int[chunkCount];
+        final int[] chunkZ = new int[chunkCount];
+        final short[] primaryBitMask = new short[chunkCount];
+        final short[] additionalBitMask = new short[chunkCount];
 
-		for (int i = 0; i < chunkCount; i++) {
-			final Chunk chunk = chunks[i];
-			Pair<byte[], Short> chunkData;
-			try {
-				chunkData = ChunkType1_7_6.serialize(chunk);
-				output.write(chunkData.key());
-			} catch (Exception e) {
-				throw new RuntimeException("Unable to serialize chunk", e);
-			}
-			chunkX[i] = chunk.getX();
-			chunkZ[i] = chunk.getZ();
-			primaryBitMask[i] = (short) chunk.getBitmask();
-			additionalBitMask[i] = chunkData.value();
-		}
-		final byte[] data = output.toByteArray();
+        for (int i = 0; i < chunkCount; i++) {
+            final Chunk chunk = chunks[i];
+            Pair<byte[], Short> chunkData;
+            try {
+                chunkData = ChunkType1_7_6.serialize(chunk);
+                output.write(chunkData.key());
+            } catch (Exception e) {
+                throw new RuntimeException("Unable to serialize chunk", e);
+            }
+            chunkX[i] = chunk.getX();
+            chunkZ[i] = chunk.getZ();
+            primaryBitMask[i] = (short) (chunk.getBitmask() & 0xFFFF); // Clamp to 16 sections
+            additionalBitMask[i] = chunkData.value();
+        }
+        final byte[] data = output.toByteArray();
 
-		final Deflater deflater = new Deflater();
-		byte[] compressedData;
-		int compressedSize;
-		try {
-			deflater.setInput(data, 0, data.length);
-			deflater.finish();
-			compressedData = new byte[data.length];
-			compressedSize = deflater.deflate(compressedData);
-		} finally {
-			deflater.end();
-		}
+        final Deflater deflater = new Deflater();
+        byte[] compressedData;
+        int compressedSize;
+        try {
+            deflater.setInput(data, 0, data.length);
+            deflater.finish();
+            compressedData = new byte[data.length];
+            compressedSize = deflater.deflate(compressedData);
+        } finally {
+            deflater.end();
+        }
 
-		byteBuf.writeShort(chunkCount);
-		byteBuf.writeInt(compressedSize);
+        byteBuf.writeShort(chunkCount);
+        byteBuf.writeInt(compressedSize);
 
-		boolean skyLight = false;
-		for (Chunk chunk : chunks) {
-			for (ChunkSection section : chunk.getSections()) {
-				if (section != null && section.getLight().hasSkyLight()) {
-					skyLight = true;
-					break;
-				}
-			}
-		}
+        byteBuf.writeBoolean(true); // Force sky light to true for 1.8 compatibility
```

Comment:
Why? skylight also exists in 1.7 (as otherwise wouldn't be part of the chunk data) so this is most likely a workaround for a server issue you have or something in an upper protocol breaking